### PR TITLE
Add rand feature flag to the example documentation

### PR DESCRIPTION
**Pull request by yancyribbens**
It's confusing trying to follow the documentation [here](https://docs.rs/bitcoin/latest/bitcoin/util/address/index.html) unless you know to enable the `rand` feature flag.  This PR updates the docs so people know to enable the flag.

```
22:35 < control> hello, how can i generate a simple wallet address using rust bitcoin? cant find working example
22:45 < andytoshi> control: do you have a private key?
22:45 < andytoshi> what kind of address do you want to generate?
22:46 < andytoshi> my guess would be that you want to use bdk rather than rust-bitcoin directl
22:48 < control> P2PKH address and generate private key. just as simple as bit library in python does
22:52 < control> im trying to run this example
22:52 < control> https://docs.rs/bitcoin/latest/bitcoin/util/address/index.html
22:52 < control> but it says ^^^^ could not find `rand` in `secp256k1`
```

### CODE REVIEW

**Review by Kixunil — APPROVED** [review]
ACK 50a665d966f95b310353a91d11da6a7da24261f2

File: bitcoin/src/address.rs [hunk 1]
```diff
@@ -21,6 +21,12 @@
 //! // Generate pay-to-pubkey-hash address.
 //! let address = Address::p2pkh(&public_key, Network::Bitcoin);
 //! ```
+//!
+//! # Note: creating a new address requires the rand feature flag
+//!
+//! ```rust
+//! bitcoin = { version = "*", features = ["rand"] }
```
Comment (yancyribbens) [inline bitcoin/src/address.rs:28]: @TheBlueMatt good point - the wild card operator works better here.

**Review by tcharding — DISMISSED** [review]
ACK 28fdbbaba8d201bd343c875bff36302d07b164ad

**Comment by tcharding** [comment]
Thanks for the contribution man! I built the docs and verified that it looks good in HTML.

File: bitcoin/src/address.rs (hunk 1 above)
Comment (Kixunil) [inline bitcoin/src/address.rs:28]: Please, don't use wildcard here - it'd tempt people into using it in real code which is a bad idea. I suggest using `...`.
Also it's not `rust` but `toml`.

**Review by Kixunil — CHANGES_REQUESTED** [review]

**Review by apoelstra — APPROVED** [review]
ACK f7a6d171438bd97b371efcc9adb01d24b0f345cd

**Review by Kixunil — APPROVED** [review]
ACK f7a6d171438bd97b371efcc9adb01d24b0f345cd

**Pull request merged**